integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/noarkexchange/IntegrajonspunktReceiveImpl.java

```diff
@@ -213,6 +224,24 @@ public EDUCore convertAsicEntrytoEduDocument(byte[] bytes) throws MessageExcepti
                     return unMarshaller.unmarshal(new StreamSource(zipInputStream), EDUCore.class).getValue();
                 }
             }
+        } catch (IOException | JAXBException e) {
+            throw new MessageException(StatusMessage.UNABLE_TO_EXTRACT_BEST_EDU);
+        }
+        throw new MessageException(StatusMessage.UNABLE_TO_EXTRACT_BEST_EDU);
+    }
+
+    public Arkivmelding convertAsicEntryToArkivmelding(byte[] bytes) throws MessageException {
+        try (ZipInputStream zipInputStream = new ZipInputStream(new ByteArrayInputStream(bytes))) {
+            ZipEntry entry;
+            while ((entry = zipInputStream.getNextEntry()) != null) {
+                if ("arkivmelding.xml".equals(entry.getName())) {
+                    JAXBContext jaxbContext = JAXBContext.newInstance(Arkivmelding.class);
+                    Unmarshaller unMarshaller = jaxbContext.createUnmarshaller();
+                    return unMarshaller.unmarshal(new StreamSource(zipInputStream), Arkivmelding.class).getValue();
+                }
+            }
+        } catch (IOException | JAXBException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1166)